src/compiler/parser.ts

```diff
@@ -4537,7 +4543,15 @@ module ts {
             else {
                 node.members = createMissingList<ClassElement>();
             }
-            return finishNode(node);
+
+            var finishedNode = finishNode(node);
+            if (languageVersion >= ScriptTarget.ES6) {
+                setStrictModeContext(savedStrictModeContext);

```

Comment:
So it will be a breaking change if we always set it regardless of the target. I am fine with that. However, I think we may want to discuss about it more. I would rather have a less strict version first and then tighten it later